File: src/main/java/com/github/liosha2007/android/library/activity/view/BaseActivityView.java
package com.github.liosha2007.android.library.activity.view;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import com.github.liosha2007.android.library.activity.controller.BaseActivityController;
import com.github.liosha2007.android.library.common.Utils;

/**
 * @author Aleksey Permyakov
 */
public abstract class BaseActivityView<T extends BaseActivityController> {
    protected View view;
    protected T controller;
    private int layoutId;

    public BaseActivityView(int layoutId) {
        this.layoutId = layoutId;
    }

    public void onCreate() {
        controller.setContentView(layoutId);
        view = controller.findViewById(android.R.id.content);
    }

    /**
     * Get View by ID
     *
     * @param id
     * @param <T>
     * @return
     */
    public <T extends View> T view(int id) {
        if (view == null) {
            Log.e(Utils.TAG, "view is null");
            return null;
        }
        return (T) Utils.view(view, id);
    }

    /**
     * Inflate layout
     * @param id
     * @param <T>
     * @return
     */
    public <T extends View> T inflate(int id) {
        return (T) controller.getLayoutInflater().inflate(id, null);
    }

    /**
     * Will be called when application should save data
     *
     * @param storage to save application data
     */
    public void onSaveData(Bundle storage) {
        storage.putString("WORKAROUND_FOR_BUG_19917_KEY", "WORKAROUND_FOR_BUG_19917_VALUE");
    }

    /**
     * Will be called when application should restore data
     *
     * @param storage to restore data
     */
    public void onRestoreData(Bundle storage) {
    }

    public void setController(T controller) {
        this.controller = controller;
    }
}

File: android-library/src/main/kotlin/com/x256n/android/library/view/BaseActivityView.kt
package com.x256n.android.library.view

import com.x256n.android.library.controller.BaseActivityController
import android.view.View
import com.x256n.android.library.common.Utils
import android.os.Bundle
import com.x256n.android.library.controller
import com.x256n.android.library.controller.ActivityController
import android.app.Activity

/**
 * @author Aleksey Permyakov (13.02.2015)
 */
public abstract class BaseActivityView<C : Activity> {
    protected abstract val controller: C
    protected abstract val menuLayoutId: Int
    protected abstract val layoutId: Int
    protected abstract var view: View
    public fun <T : View> view(viewId: Int?): T? = Utils.view(view, viewId)
    public fun <T : View> inflate(viewId: Int): T? = controller.getLayoutInflater().inflate(viewId, null) as? T
}